test_conformance/images/kernel_image_methods/test_3D.cpp

```diff
@@ -113,23 +116,26 @@ int test_get_image_info_3D(cl_device_id device, cl_context context,
                 imageInfo.height = (size_t)random_log_in_range( 16, (int)maxHeight / 32, seed );
                 imageInfo.depth = (size_t)random_log_in_range( 16, (int)maxDepth / 32, seed );
 
-                imageInfo.rowPitch = imageInfo.width * pixelSize;
-                imageInfo.slicePitch = imageInfo.rowPitch * imageInfo.height;
+                rowPitch = imageInfo.width * pixelSize;
+                slicePitch = imageInfo.rowPitch * imageInfo.height;
 
                 size_t extraWidth = (int)random_log_in_range( 0, 64, seed );
-                imageInfo.rowPitch += extraWidth;
+                rowPitch += extraWidth;
 
                 do {
                     extraWidth++;
-                    imageInfo.rowPitch += extraWidth;
-                } while ((imageInfo.rowPitch % pixelSize) != 0);
+                    rowPitch += extraWidth;
+                } while ((rowPitch % pixelSize) != 0);
 
                 size_t extraHeight = (int)random_log_in_range( 0, 8, seed );
-                imageInfo.slicePitch = imageInfo.rowPitch * (imageInfo.height + extraHeight);
+                slicePitch = rowPitch * (imageInfo.height + extraHeight);
 
-                size = (cl_ulong)imageInfo.slicePitch * (cl_ulong)imageInfo.depth * 4 * 4;
+                size = (cl_ulong)slicePitch * (cl_ulong)imageInfo.depth * 4 * 4;
             } while(  size > maxAllocSize || ( size * 3 ) > memSize );
 
+            imageInfo.slicePitch = slicePitch;
```

Comment:
I'm not sure I understand the fix. Aren't we just moving the overflow possibility to here? It seems like a fundamental limitation of the API that rowPitch and slicePitch have to fit in size_t.

Reply:
The trick here is that for a 32-bit system, if the pitch is more than 32-bit then this `while(  size > maxAllocSize || ( size * 3 ) > memSize );` will cause it to recalculate as maxAllocSize and memSize  are guaranteed to be <=4GB. So if we get to this point outside the loop the pitches are guaranteed to fit.

Reply:
Thanks, got it. I was looking at the wrong loop scope.